chips/esp32-c3/src/timg.rs

```diff
@@ -0,0 +1,244 @@
+    unsafe { StaticRef::new(0x6002_0000 as *const TimgRegisters) };
+
+/// 80MHz `Frequency`
+#[derive(Debug)]
+pub struct Freq20MHz;
+impl time::Frequency for Freq20MHz {
+    fn frequency() -> u32 {
+        20_000_000
+    }
+}
+
+register_structs! {
+    pub TimgRegisters {
+        (0x000 => t0config: ReadWrite<u32, CONFIG::Register>),
+        (0x004 => t0lo: ReadWrite<u32>),
+        (0x008 => t0hi: ReadWrite<u32>),
+        (0x00C => t0update: ReadWrite<u32>),
+        (0x010 => t0alarmlo: ReadWrite<u32>),
+        (0x014 => t0alarmhi: ReadWrite<u32>),
+        (0x018 => t0loadlo: ReadWrite<u32>),
+        (0x01C => t0loadhi: ReadWrite<u32>),
+        (0x020 => t0load: ReadWrite<u32>),
+
+        (0x024 => _reserved0),
+
+        (0x048 => wdtconfig0: ReadWrite<u32, WDTCONFIG0::Register>),
+        (0x04C => wdtconfig1: ReadWrite<u32, WDTCONFIG1::Register>),
+        (0x050 => wdtconfig2: ReadWrite<u32>),
+        (0x054 => wdtconfig3: ReadWrite<u32>),
+        (0x058 => wdtconfig4: ReadWrite<u32>),
+        (0x05C => wdtconfig5: ReadWrite<u32>),
+        (0x060 => wdtfeed: ReadWrite<u32>),
+        (0x064 => wdtwprotect: ReadWrite<u32>),
+
+        (0x068 => rtccalicfg: ReadWrite<u32, RTCCALICFG::Register>),
+        (0x06C => rtccalicfg1: ReadWrite<u32, RTCCALICFG1::Register>),
+
+        (0x070 => int_ena: ReadWrite<u32, INT::Register>),
+        (0x074 => int_raw: ReadWrite<u32, INT::Register>),
+        (0x078 => int_st: ReadWrite<u32, INT::Register>),
+        (0x07C => int_clr: ReadWrite<u32, INT::Register>),
+        (0x080 => @END),
+    }
+}
+
+register_bitfields![u32,
+    CONFIG [
+        USE_XTAL OFFSET(9) NUMBITS(1) [],
+        ALARM_EN OFFSET(10) NUMBITS(1) [],
+        LEVEL_INT_EN OFFSET(11) NUMBITS(1) [],
+        DIVIDER_RST OFFSET(12) NUMBITS(1) [],
+        DIVIDER OFFSET(13) NUMBITS(16) [],
+        AUTORELOAD OFFSET(29) NUMBITS(1) [],
+        INCREASE OFFSET(30) NUMBITS(1) [],
+        EN OFFSET(31) NUMBITS(1) [],
+    ],
+    WDTCONFIG0 [
+        FLASHBOOT_MOD_EN OFFSET(14) NUMBITS(1) [],
+        SYS_RESET_LENGTH OFFSET(15) NUMBITS(3) [],
+        CPU_RESET_LENGTH OFFSET(18) NUMBITS(3) [],
+        LEVEL_INT_EN OFFSET(21) NUMBITS(1) [],
+        EDGE_INT_EN OFFSET(22) NUMBITS(1) [],
+        STG3 OFFSET(23) NUMBITS(2) [],
+        STG2 OFFSET(25) NUMBITS(2) [],
+        STG1 OFFSET(27) NUMBITS(2) [],
+        STG0 OFFSET(29) NUMBITS(2) [],
+        EN OFFSET(31) NUMBITS(1) [],
+    ],
+    WDTCONFIG1 [
+        CLK_PRESCALE OFFSET(16) NUMBITS(16) [],
+    ],
+    RTCCALICFG [
+        START_CYCLING OFFSET(12) NUMBITS(1) [],
+        CLK_SEL OFFSET(13) NUMBITS(2) [],
+        RDY OFFSET(15) NUMBITS(1) [],
+        MAX OFFSET(16) NUMBITS(15) [],
+        START OFFSET(31) NUMBITS(1) [],
+    ],
+    RTCCALICFG1 [
+        VALUE OFFSET(7) NUMBITS(25) [],
+    ],
+    INT [
+        T0 OFFSET(0) NUMBITS(1) [],
+        WDT OFFSET(1) NUMBITS(1) [],
+    ],
+];
+
+#[repr(u8)]
```

Comment:
Same here

Reply:
You are right, I modified it to u32.

Reply:
My inclination for all of these cases would be to simply not specify the representation at all -- these types are not used when reading or writing directly to registers or across any other FFI boundary, so why do we need to specify how they are represented in memory?

Reply:
I has the impression that this was needed to automatically cast to u32, but it is not. I deleted the repr.